cargo-dylint/tests/ci.rs

```diff
@@ -713,6 +715,103 @@ fn update() {
     });
 }
 
+#[test]
+fn lint_script_test() {
+    preserves_cleanliness("lint_script_test", false, || {
+        let cargo_dylint_path = Path::new("target/debug/cargo-dylint")
+            .canonicalize()
+            .expect("Failed to find cargo-dylint executable. Ensure it is built, e.g., with `cargo build -p cargo-dylint`.");
+
+        let example_restriction_dir = Path::new("examples/restriction");
+        let mut restriction_libs = Vec::new();
+        if example_restriction_dir.is_dir() {
+            for entry in read_dir(example_restriction_dir).unwrap() {
```

Comment:
```suggestion
            for entry in read_dir("examples/restriction").unwrap() {
```
I think this should work, eliminating the need for `example_restriction_dir`.